worldedit-bukkit/src/main/java/com/sk89q/worldedit/bukkit/WorldEditListener.java

```diff
@@ -124,31 +144,36 @@ public void onPlayerInteract(PlayerInteractEvent event) {
             }
 
         } else if (action == Action.LEFT_CLICK_AIR) {
-
-            if (we.handleArmSwing(player)) {
+            if (!isDuplicateInteraction(player) && we.handleArmSwing(player)) {
```

Comment:
I used to deduplicate all kind of clicks but during tests I noticed that if I both left click and right click at the same time, I could break blocks while holding a wooden axe. The fix here is to never deduplicate `left click on block` events to ensure they are always cancelled when needed. 

We don't need to deduplicate `left click on block` events because they always occurs before the `click on air` events. As long as we deduplicate `click on air` events, this is fine.

Some events need to be deduplicated, some not, that's why the filter is not on the top.

Reply:
Hmm, that makes me wonder if this is fully correct. Perhaps we should also cache the cancellation for the last result and repeat it, then?

Reply:
Yes that would make sense. I'll try and see what the implications of this are.